src/components/HyperlinkingWrapper.js

```diff
@@ -72,17 +75,29 @@ export default class HyperlinkingWrapper extends Component {
                         range.index + range.length - index,
                     );
 
-                    currentSelection = this.quill.getBounds(new Range(index, length));
+                    selectionBounds = this.quill.getBounds(new Range(index, length));
                 }
 
-                this.setState({ currentSelection });
+                this.setState({ selectionBounds, selectionFormat });
                 this.quill.format('highlight', true);
+            } else {
+                const [blot, blotRange] = this.getBlotFromIndex(range.index);
+                const allowAutoselect = AUTOSELECT_BLOT_TYPES.indexOf(blot.statics.blotName) !== -1;
+
+                if (allowAutoselect) {
+                    this.quill.setSelection(blotRange);
+                }
             }
         }
     }
 
     onClose() {
-        this.setState({ currentSelection: null, isLinkInvalid: false, current: HYPERLINKING_STATE.INITIAL });
+        this.setState({
+            selectionBounds: null,
+            selectionFormat: {},
```

Comment:
can we have a null instead?

Reply:
when no formatting is available, the default value returned by quill's getFormat is `{}`, so I would rather stick to that. If you have any objections, let me know.